tests/unit/test_charm.py

```diff
@@ -429,6 +433,20 @@ def bundle_handler(args: ops.testing.ExecArgs) -> None:
     charm._on_anonymize_user_action(event)
 
 
+def test_handle_pebble_ready_event():
+    """
+    arrange: given a deployed discourse charm with all the required relations
+    act: trigger the pebble ready event on a leader unit
+    assert: the pebble plan gets updated
+    """
+    harness = helpers.start_harness()
+
+    plan_before_event = harness.get_container_pebble_plan(CONTAINER_NAME)
+    harness.container_pebble_ready(CONTAINER_NAME)
+    plan_after_event = harness.get_container_pebble_plan(CONTAINER_NAME)
+    assert plan_before_event.__dict__ != plan_after_event.__dict__
```

Comment:
This just checks that the dictionary has changed, do we know that it has changed in the way we expect? E.g., the service was modified as expected?

Reply:
I've opted to add two asserts: one to test that `_services` is in the `__dict__` and one to assert that `discourse` is in `__dict__["_services"]`. I don't think that going further and checking all the properties is meaningful. Do you agree ?

Reply:
Looks good, thanks